### ISSUE TITLE
verbatimModuleSyntax = true switches incremental compilation to isolatedModules = true

### ISSUE THREAD

**Issue by uasan** [microsoft/TypeScript#53374]
Hello, the new option `verbatimModuleSyntax` has a very simple and clear description, but nothing is said here about how this option affects incremental compilation.

```JS
compilerOptions = {
  incremental: true,
  isolatedModules: false,
  verbatimModuleSyntax: true,
}

```

When this option is enabled, when changing a file on which other files depend (they import this file), these dependent files are not added to the set of changed files, the same effect will be if you enable the `isolatedModules = true` option.

I wanted to know if this is intended behavior or a bug?
Not including dependent files in the set of changed files breaks some logic in our builder.

**Comment by fatcerberus** [comment]
`isolatedModules: false, verbatimModuleSyntax: true` is not a valid combination - the latter implies the former.  In fact this set of options should produce an error to the effect that both of those options can't be specified at the same time.

**Comment by uasan** [comment]
Ok, then you need to give an error with this combination of settings

**Comment by uasan** [comment]
unfortunately in version TS 5.0.2 i don't get this error

**Comment by fatcerberus** [comment]
Huh.  Guess there is a bug, then, just not the one you thought it was 😄

**Comment by JoshuaKGoldberg** [comment]
Err, I get this locally with TS 5.0.2?

```plaintext
$ cat tsconfig.json
{
  "compilerOptions": {
    "incremental": true,
    "isolatedModules": true,
    "verbatimModuleSyntax": true
  }
}
```

```plaintext
$ tsc --version
Version 5.0.2
```

```plaintext
$ tsc
tsconfig.json:4:5 - error TS5104: Option 'isolatedModules' is redundant and cannot be specified with option 'verbatimModuleSyntax'.

4     "isolatedModules": true,
      ~~~~~~~~~~~~~~~~~


Found 1 error in tsconfig.json:4
```
